test/test_reactant.jl

```diff
@@ -61,7 +61,109 @@ GPUArrays.allowscalar(true)
+
+    set!(c, (x, y, z) -> x + y * z)
+    x, y, z = nodes(c)
+
+    @allowscalar begin
+        @test c[1, 1, 1] == x[1] + y[1] * z[1]
+        @test c[1, 2, 1] == x[1] + y[2] * z[1]
+        @test c[1, 2, 3] == x[1] + y[2] * z[3]
+    end
+
+    @jit fill_halo_regions!(c)
+
+    @allowscalar begin
+        @test c[1, 1, 0] == c[1, 1, 1]
+    end
+
+    d = CenterField(grid)
+    parent(d) .= 2
+
+    cd = Field(c * d)
+    compute!(cd)
+
+    @allowscalar begin
+        @test cd[1, 1, 1] == 2 * (x[1] + y[1] * z[1])
+        @test cd[1, 2, 1] == 2 * (x[1] + y[2] * z[1])
+        @test cd[1, 2, 3] == 2 * (x[1] + y[2] * z[3])
+    end
+end
+
+@testset "Reactant Super Simple Simulation Tests" begin
+    nonhydrostatic_model_kw = (; advection=WENO())
+    hydrostatic_model_kw = (; momentum_advection=WENO())
+    Nx, Ny, Nz = (10, 10, 10) # number of cells
+    halo = (7, 7, 7)
+    longitude = (0, 4)
+    latitude = (0, 4)
+    z = (-1, 0)
+    lat_lon_kw = (; size=(Nx, Ny, Nz), halo, longitude, latitude, z)
+    rectilinear_kw = (; size=(Nx, Ny, Nz), halo, x=(0, 1), y=(0, 1), z=(0, 1))
+
+    # FFTs are not supported by Reactant so we don't run this test:
+    # @info "Testing RectilinearGrid + NonhydrostaticModel Reactant correctness"
+    # test_reactant_model_correctness(RectilinearGrid, NonhydrostaticModel, rectilinear_kw, nonhydrostatic_model_kw)
+
+    @info "Testing RectilinearGrid + HydrostaticFreeSurfaceModel Reactant correctness"
+    hydrostatic_model_kw = (; free_surface=ExplicitFreeSurface(gravitational_acceleration=1))
+    test_reactant_model_correctness(RectilinearGrid, HydrostaticFreeSurfaceModel, rectilinear_kw, hydrostatic_model_kw)
+
+    @info "Testing LatitudeLongitudeGrid + HydrostaticFreeSurfaceModel Reactant correctness"
+    hydrostatic_model_kw = (; momentum_advection=WENO())
+    test_reactant_model_correctness(LatitudeLongitudeGrid, HydrostaticFreeSurfaceModel, lat_lon_kw, hydrostatic_model_kw)
+
+    #=
+    equation_of_state = TEOS10EquationOfState()
+    hydrostatic_model_kw = (momentum_advection = WENOVectorInvariant(),
+                            tracer_advection = WENO(),
+                            tracers = (:T, :S, :e),
+                            buoyancy = SeawaterBuoyancy(; equation_of_state),
+                            closure = CATKEVerticalDiffusivity())
+    test_reactant_model_correctness(LatitudeLongitudeGrid, HydrostaticFreeSurfaceModel, lat_lon_kw, hydrostatic_model_kw)
+    =#
 end
+
+@testset "Reactant test traced clock" begin
+    @info "Testing ConcreteRNumber clock elements in a model with time stepping..."
+    halo = (7, 7, 7)
+
+    # All of these may not need to be traced but this is paranoia.
+    FT = Float64
+    time = ConcreteRNumber(zero(FT))
+    iteration = ConcreteRNumber(0)
+    stage = ConcreteRNumber(0)
+    last_Δt = ConcreteRNumber(zero(FT))
+    last_stage_Δt = ConcreteRNumber(zero(FT))
+    clock = Clock{Float64}(; time, iteration, stage, last_Δt, last_stage_Δt)
```

Comment:
@wsmoses these lines show how to implement a Clock with ConcreteRNumber I think.

Reply:
are we sure this shouldn't be Clock{ConcreteRNumber{Float64}} ?

Reply:
grr you're right

Reply:
ok fixed